maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java

```diff
@@ -422,6 +439,106 @@ public String toString() {
         }
     }
 
+    /**
+     * Represents a combination in the version item list.
+     * It is usually a combination of a string and a number, with the string first and the number second
+     */
+    private static class CombinationItem implements Item {
+
+        StringItem stringValue;
```

Comment:
This confused since I initially thought these were different representations of the same value. How about stringPart and digitPart?

Reply:
That's a good idea. I changed it.